### PR TITLE
Fix nodegroup creation when imageBuilder is enabled, fix duplicate policies

### LINKED ISSUE #1696: managedNodeGroups fails with imageBuilder
### DESCRIPTION 

When you create a new cluster with `managedNodeGroups` and set the `iam.withAddonPolicies.imageBuilder` to `true`, the `Nodegroup/ManagedNodeGroup` will fail, causing a ROLLBACK.

### STEPS

```bash
eksctl create cluster -f cluster.yaml
```

```yaml
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: test3-cluster
  region: us-east-2

managedNodeGroups:
  - name: test3-workers
    minSize: 2
    maxSize: 4
    desiredCapacity: 3
    labels: {role: worker}
    tags:
      nodegroup-role: worker
    iam:
      withAddonPolicies:
        externalDNS: true
        certManager: true
        imageBuilder: true
```

### EXPECTED
This not to fail.

### ACTUAL 

```
[ℹ]  eksctl version 0.11.1
[ℹ]  using region us-east-2
[ℹ]  setting availability zones to [us-east-2c us-east-2a us-east-2b]
[ℹ]  subnets for us-east-2c - public:192.168.0.0/19 private:192.168.96.0/19
[ℹ]  subnets for us-east-2a - public:192.168.32.0/19 private:192.168.128.0/19
[ℹ]  subnets for us-east-2b - public:192.168.64.0/19 private:192.168.160.0/19
[ℹ]  using Kubernetes version 1.14
[ℹ]  creating EKS cluster "test3-cluster" in "us-east-2" region with managed nodes
[ℹ]  1 nodegroup (test3-workers) was included (based on the include/exclude rules)
[ℹ]  will create a CloudFormation stack for cluster itself and 0 nodegroup stack(s)
[ℹ]  will create a CloudFormation stack for cluster itself and 1 managed nodegroup stack(s)
[ℹ]  if you encounter any issues, check CloudFormation console or try 'eksctl utils describe-stacks --region=us-east-2 --cluster=test3-cluster'
[ℹ]  CloudWatch logging will not be enabled for cluster "test3-cluster" in "us-east-2"
[ℹ]  you can enable it with 'eksctl utils update-cluster-logging --region=us-east-2 --cluster=test3-cluster'
[ℹ]  Kubernetes API endpoint access will use default of {publicAccess=true, privateAccess=false} for cluster "test3-cluster" in "us-east-2"
[ℹ]  2 sequential tasks: { create cluster control plane "test3-cluster", create managed nodegroup "test3-workers" }
[ℹ]  building cluster stack "eksctl-test3-cluster-cluster"
[ℹ]  deploying stack "eksctl-test3-cluster-cluster"
[ℹ]  building managed nodegroup stack "eksctl-test3-cluster-nodegroup-test3-workers"
[ℹ]  deploying stack "eksctl-test3-cluster-nodegroup-test3-workers"
[✖]  unexpected status "ROLLBACK_IN_PROGRESS" while waiting for CloudFormation stack "eksctl-test3-cluster-nodegroup-test3-workers"
[ℹ]  fetching stack events in attempt to troubleshoot the root cause of the failure
[✖]  AWS::IAM::Policy/PolicyCertManagerGetChange: CREATE_FAILED – "Resource creation cancelled"
[✖]  AWS::IAM::Policy/PolicyCertManagerHostedZones: CREATE_FAILED – "Resource creation cancelled"
[✖]  AWS::IAM::Policy/PolicyCertManagerChangeSet: CREATE_FAILED – "Resource creation cancelled"
[✖]  AWS::EKS::Nodegroup/ManagedNodeGroup: CREATE_FAILED – "The provided role doesn't have the Amazon EKS Managed Policies associated with it. Please ensure the following policies [arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy, arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy, arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly] are attached (Service: AmazonEKS; Status Code: 400; Error Code: InvalidParameterException; Request ID: ccf7fea0-2d47-4a61-8930-b0a7b1a9d433)"
[ℹ]  1 error(s) occurred and cluster hasn't been created properly, you may wish to check CloudFormation console
[ℹ]  to cleanup resources, run 'eksctl delete cluster --region=us-east-2 --name=test3-cluster'
[✖]  waiting for CloudFormation stack "eksctl-test3-cluster-nodegroup-test3-workers": ResourceNotReady: failed waiting for successful resource state
[✖]  failed to create cluster "test3-cluster"
```

### VERSIONS

* Mac OS X `10.14.6`
* eksctl `0.11.1`
* kubectl `v1.17.0`

### NOTES

The same cluster definition without the `imageBuilder: true` will succeed.

### PR DESCRIPTION
**Pull request by cPu1**
### Description

Fixes #1696 

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, and `examples` directory)
- [x] Manually tested
- [x] Added labels for change area (e.g. `area/nodegroup`) and target version (e.g. `version/0.12.0`)
- [ ] Added note in `docs/release_notes/draft.md` (or relevant release note)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/alecthomas/jsonschema v0.0.0-20190530235721-fd8d96416671
 	github.com/aws/aws-sdk-go v1.25.14-0.20191218135223-757eeed07291
 	github.com/awslabs/goformation v0.0.0-20190320125420-ac0a17860cf1
```
Comment (martina-if) [inline go.mod]: Should we only use v4? and remove this one?

File: pkg/cfn/builder/managed_nodegroup_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			addons: api.NodeGroupIAMAddonPolicies{
+				ImageBuilder: api.Enabled(),
+				CloudWatch:   api.Enabled(),
+			},
+			expectedManagedPolicies: []string{"CloudWatchAgentServerPolicy", "AmazonEC2ContainerRegistryPowerUser"},
+		},
+	}
+
+	for i, tt := range iamRoleTests {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
```
Comment (martina-if) [inline pkg/cfn/builder/managed_nodegroup_test.go]: Can we have tests descriptions instead of numbers?

Comment (cPu1) [reply pkg/cfn/builder/managed_nodegroup_test.go]: Setting the description for every test case would result in some boilerplate code (and also copy-paste as some test cases somewhat address the same case). If a test case fails, the index would tell us which one failed. Moreover, the assertions itself would include some information about the failure.

File: go.mod (hunk 1 above)
Comment (cPu1) [reply go.mod]: This one gets redirected to github.com/errordeveloper/goformation (via a replace directive below), which is Ilya's fork that adds features not present upstream, so we can't replace this with upstream `goformation/v4` without changing the code.

File: pkg/cfn/builder/managed_nodegroup_test.go (hunk 2 above)
Comment (martina-if) [reply pkg/cfn/builder/managed_nodegroup_test.go]: Can you add a little text explaining what it is testing and the number as well?

Comment (cPu1) [reply pkg/cfn/builder/managed_nodegroup_test.go]: I added a little description to each test. This is how it looks like:

```
--- PASS: TestManagedResources (0.01s)
    --- PASS: TestManagedResources/0:_Default_policies (0.00s)
    --- PASS: TestManagedResources/1:_ImageBuilder_enabled (0.00s)
    --- PASS: TestManagedResources/2:_CloudWatch_enabled (0.00s)
    --- PASS: TestManagedResources/3:_Custom_policies (0.00s)
    --- PASS: TestManagedResources/4:_Custom_policies (0.00s)
    --- PASS: TestManagedResources/5:_Duplicate_policies (0.00s)
    --- PASS: TestManagedResources/6:_Multiple_duplicate_policies (0.00s)
```

Comment (martina-if) [reply pkg/cfn/builder/managed_nodegroup_test.go]: Thanks

**Pull request merged**